### PR TITLE
Add `IntervalChange`, `LocalDateChange`, `LocalTimeChange`.

### PR DESCRIPTION
**Pull request by alexander-aleksandrov**

### CODE REVIEW

File: values/src/main/java/org/spine3/time/change/Changes.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+    private Changes() {
+    }
+
+    /**
+     * Creates {@link IntervalChange} object for the passed previous and new field values of time interval.
+     *
+     * <p>Passed values cannot be equal.
+     */
+    public static IntervalChange ofInterval(Timestamp previousStartValue, Timestamp newStartValue,
```
Comment (alexander-yevsyukov) [inline values/src/main/java/org/spine3/time/change/Changes.java]: No, it must be two intervals. It's easy to make an error in such a chain of parameters of the same type.

File: values/src/main/java/org/spine3/time/LocalDates.java [hunk 2]
```diff
@@ -45,4 +46,22 @@ public static LocalDate today() {
                                           .build();
         return result;
     }
+
+    /**
+     * Obtains the date in the future after adding the specified amount of days.
```
Comment (alexander-yevsyukov) [inline values/src/main/java/org/spine3/time/LocalDates.java]: Do not describe what you do inside the method. Describe what we get as the result.

File: values/src/main/java/org/spine3/time/LocalDates.java [hunk 3]
```diff
@@ -45,4 +46,22 @@ public static LocalDate today() {
...
+     */
+    public static LocalDate inDays(int daysDelta) {
+        final Calendar calendar = Calendar.getInstance();
```
Comment (alexander-yevsyukov) [inline values/src/main/java/org/spine3/time/LocalDates.java]: How about `java.time.LocalDate.now().plusDays(daysDelta)`? And then converting it to our `LocalDate`?

File: values/src/main/java/org/spine3/time/change/package-info.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * This package contains classes and interfaces for working with field value changes.
```
Comment (alexander-yevsyukov) [inline values/src/main/java/org/spine3/time/change/package-info.java]: It's the specific package.

File: values/src/main/proto/spine/time/change/change.proto [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+import "spine/annotations.proto";
+import "spine/validate.proto";
+import "spine/time/zone_offset.proto";
+
+//
+// This file provides messages that define changes of fields in other messages.
+// The `TimeChange` messages can be use as a request to change a value (e.g. as a field of a command),
+// or as information on the change made (e.g. as a part of an event).
+//
+// For all types the following rule apply: `new_value` field must not be equal to `previous_value`.
```
Comment (alexander-yevsyukov) [inline values/src/main/proto/spine/time/change/change.proto]: We also need to say that this file is made after the “main” `change.proto` and give reference to it.

File: values/src/test/java/org/spine3/time/LocalDatesShould.java [hunk 6]
```diff
@@ -45,4 +45,13 @@ public void obtain_current_date() {
         assertEquals(calendar.get(Calendar.MONTH) + 1, today.getMonthValue());
         assertEquals(calendar.get(Calendar.DAY_OF_MONTH), today.getDay());
     }
+
+    @Test
+    public void obtain_date_in_future_after_specified_number_of_days() {
+        final LocalDate today = LocalDates.inDays(5);
+        final Calendar calendar = Calendar.getInstance();
```
Comment (alexander-yevsyukov) [inline values/src/test/java/org/spine3/time/LocalDatesShould.java]: Use `java.time`.

File: values/src/main/java/org/spine3/time/LocalDates.java [hunk 7]
```diff
@@ -45,4 +46,22 @@ public static LocalDate today() {
                                           .build();
         return result;
     }
+
+    /**
+     * Obtains the date in the future after adding the specified amount of days.
+     */
+    public static LocalDate inDays(int daysDelta) {
```
Comment (alexander-yevsyukov) [inline values/src/main/java/org/spine3/time/LocalDates.java]: Please rename to `plusDays` as in `java.time`. Also rename the names of the parameter. Review other methods and name them after counterparts from `java.time`.

File: values/src/main/java/org/spine3/time/LocalDates.java (hunk 3 above)
Comment (alexander-yevsyukov) [inline values/src/main/java/org/spine3/time/LocalDates.java]: Sorry, I forgot that we're under Java 7.

File: values/src/test/java/org/spine3/time/LocalDatesShould.java (hunk 6 above)
Comment (alexander-yevsyukov) [inline values/src/test/java/org/spine3/time/LocalDatesShould.java]: Sorry, I forgot that we're under Java 7. Please disregard.

File: values/src/main/java/org/spine3/time/LocalDates.java (hunk 2 above)
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/LocalDates.java]: Fixed

File: values/src/main/java/org/spine3/time/LocalDates.java (hunk 7 above)
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/LocalDates.java]: Fixed

File: values/src/main/java/org/spine3/time/change/package-info.java (hunk 4 above)
Comment (alexander-aleksandrov) [inline values/src/main/java/org/spine3/time/change/package-info.java]: Fixed

File: values/src/main/java/org/spine3/time/change/package-info.java [hunk 8]
```diff
@@ -19,7 +19,7 @@
  */
 
 /**
- * This package contains classes and interfaces for working with field value changes.
+ * This package contains classes and interfaces for working with Time value changes.
```
Comment (alexander-yevsyukov) [inline values/src/main/java/org/spine3/time/change/package-info.java]: Lowercase would do.

**Comment by alexander-yevsyukov** [comment]
LGTM. Please merge.

**Pull request merged**